doc/src/explanation/active-deprecations.md

```diff
@@ -0,0 +1,1231 @@
+feature is deprecated. It is problematic for a few reasons:
+
+- It can affect performance in major ways. See for instance
+  https://github.com/sympy/sympy/issues/18056 and
+  https://github.com/sympy/sympy/issues/15416 where it's caused up to 100x
+  slowdowns. The issue is that whenever a function, and hence `sympify` is
+  passed something it doesn't know how to `sympify`, for instance, a Python
+  function type, it passes the string to {func}`~.parse_expr`. This is
+  significantly slower than the direct conversions that happen by default.
+  This occurs specifically whenever `sympify()` is used in library code
+  instead of `_sympify()` (or equivalently `sympify(strict=True)`), but
+  presently this is done a lot. Using `strict=True` will at some point be the
+  default for all library code, but this is a [harder change to
+  make](https://github.com/sympy/sympy/issues/11003).
+
+- It can cause security issues, since strings are evaled, and objects can
+  return whatever string they want in their `__repr__`. See also
+  https://github.com/sympy/sympy/pull/12524.
+
+- It really isn't very useful to begin with. Just because an object's string
+  form can be parsed into a SymPy expression doesn't mean it should be parsed
+  that way. This is usually correct for custom numeric types, but an object's
+  repr could be anything. For instance, if the string form of an object looks
+  like a valid Python identifier, it will parse as a `Symbol`.
+
+There are plenty of ways to make custom objects work inside of
+{func}`~.sympify`.
+
+- Firstly, if an object is intended to work alongside other SymPy expressions,
+  it should subclass from {class}`~.Basic` (or {class}`~.Expr`). If it does,
+  {func}`~.sympify` will just return it unchanged because it will already be a
+  valid SymPy object.
+
+- For objects that you control, you can add the `_sympy_` method. The [sympify
+  docstring](sympy.core.sympify.sympify) has an example of this.
+
+- For objects that you don't control, you can add a custom converter to the
+  `sympy.core.sympify.converter` dictionary. The {func}`~.sympify` docstring
+  also has an example of this.
+
+To silence this deprecation warning in all cases, you can pass `strict=True`
+to `sympify()`. However, note that this will also disable some other
+conversions such as conversion of strings (for converting strings to SymPy
+types, you can explicitly use {func}`~.parse_expr`).
+
+(deprecated-indefinite-integral-eq)=
+### Creating an indefinite `Integral` with an `Eq` argument
+
+Passing an [`Eq()`](sympy.core.relational.Equality) object to
+{func}`~.integrate` is deprecated in the case where the integral is
+indefinite. This is because if $a = b$, $\int a\,dx = \int b\,dx$ is not true
+in general, due to the arbitrary constants (which `integrate` does not
+include).
+
+If you want to make an equality of indefinite integrals, use `Eq(integrate(a, x),
+integrate(b, x))` explicitly.
```

Comment:
Maybe also worth mentioning that if you have an equality already then you can do
```
Eq(integrate(eq.lhs, x), integrate(eq.rhs, x))
```